### ISSUE TITLE
Mahnung Funktion implementieren

### ISSUE THREAD

**Issue by jzakotnik** [jzakotnik/openlibry#144]

**Comment by fb4thxUall** [comment]
Spannend, auf meiner Feedbackliste des Bücherei-Teams steht genau das noch. Habs bisher noch nicht zum Thema gemacht, aber hier sehe ich auch Bedarf und hab schon ein paar Ideen:

Man könnte in die Richtung gehen, dass man Listen pro Klasse erstellen kann, sodass die Klassenleitung eine Liste bekommt und sich weiter kümmern kann.

Man könnte in die Richtung gehen, dass automatisch individuelle Briefe formuliert werden oder alternativ könnte man mit automatischen Mails arbeiten, wofür man aber auch eine Mailadresse hinterlegen müsste. Damit wiederum man man einen anderen Rattenschwanz auf (DSGVO usw), der scheinbare Komfortgewinn wäre also gut abzuwägen.



Am 23. September 2024 22:15:40 MESZ schrieb Jure Zakotnik ***@***.***>:
>
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/jzakotnik/openlibry/issues/144
>You are receiving this because you are subscribed to this thread.
>
>Message ID: ***@***.***>
-- 
Diese Nachricht wurde von meinem Android-Gerät mit K-9 Mail gesendet.

**Comment by jzakotnik** [comment]
@fb4thxUall ja genau in die Richtung ging es hier auch. Wollte das Feature mal vorziehen, weil man damit das Templating mit word Files und Export in pdf ganz gut üben kann - hab für den Mahnbrief schon ein Template bekommen. 

Mein Traum wäre ja, dass man neben dem fehlenden Buch noch das Buchcover drucken kann, manchmal kann man sich besser erinnern wo ein Buch rumliegt wenn man weiß wie es aussieht 📚 .

**Comment by fb4thxUall** [comment]
Ja, das ist natürlich eine coole Funktion mit dem Cover. Sofern man es denn überhaupt als Datei zur Verfügung hat :-D


Am 24. September 2024 08:52:18 MESZ schrieb Jure Zakotnik ***@***.***>:
>@fb4thxUall ja genau in die Richtung ging es hier auch. Wollte das Feature mal vorziehen, weil man damit das Templating mit word Files und Export in pdf ganz gut üben kann - hab für den Mahnbrief schon ein Template bekommen. 
>
>Mein Traum wäre ja, dass man neben dem fehlenden Buch noch das Buchcover drucken kann, manchmal kann man sich besser erinnern wo ein Buch rumliegt wenn man weiß wie es aussieht 📚 . 
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/jzakotnik/openlibry/issues/144#issuecomment-2370342312
>You are receiving this because you were mentioned.
>
>Message ID: ***@***.***>
-- 
Diese Nachricht wurde von meinem Android-Gerät mit K-9 Mail gesendet.

**Comment by jzakotnik** [comment]
Das ist ein Stück weiter jetzt. Aktuell erzeugt es noch Office XML, was aber mit LibreOffice und Word gleichermaßen zu öffnen ist. 
Frage ist noch wie man rausbekommt wie hoch die Mahngebühren sind?
<img width="423" alt="image" src="https://github.com/user-attachments/assets/658e270f-59f0-4b49-97f5-1742c436aafa">

<img width="583" alt="image" src="https://github.com/user-attachments/assets/3f38bb45-9206-4074-889f-31b131027340">

**Comment by fb4thxUall** [comment]
Mahngebühren könnten ja in der .env hinterlegt sein. 

Könntest du dein Beispielword zur Verfügung stellen? Ich würde auch dafür plädieren, es standardmäßig mit auszuliefern. Denn dann hat man direkt eine Vorlage und muss diese nur noch anpassen. Ohne technisches Know-how. Ich weiß selbst übrigens spontan nämlich auch nicht, wie ich das erstellen müsste, damit das funktioniert. :-)

**Comment by jzakotnik** [comment]
@fb4thxUall das Beispiel Template sollte hoffenltich in public/mahnung-template.docx eingecheckt sein, findest Du es da?

**Comment by fb4thxUall** [comment]
Ja... in dem Ordner hatte ich vorhin sogar geschaut. Wer lesen kann... 

Vielen Dank

**Comment by fb4thxUall** [comment]
Hab es jetzt das erste Mal testen können. Es entsteht wie erwartet ein .docx, allerdings ist das komplett leer. Fehler gibt es nicht wirklich. In der Konsole steht:  

`Printing reminder letters via API
Rentals {}
Variables for docxtemplater { alleMahnungen: [] }
Generated doc <Buffer 50 4b 03 04 0a 00 00 00 08 00 00 00 21 00 1e 91 1a b7 e9 00 00 00 4e 02 00 00 0b 00 00 00 5f 72 65 6c 73 2f 2e 72 65 6c 73 ad 92 c1 6a c3 30 0c 40 ef ... 13335 more bytes>
`

Konfiguration im Test war alles default aus der .env_example und das mitgelieferte Beispieltemplate.
